modules/iam-role-for-service-accounts-eks/policies.tf

```diff
@@ -531,24 +531,23 @@ data "aws_iam_policy_document" "karpenter_controller" {
       "ec2:DescribeInstanceTypeOfferings",
       "ec2:DescribeAvailabilityZones",
       "ec2:DescribeSpotPriceHistory",
+      "ec2:RunInstances",
+      "iam:PassRole",
       "pricing:GetProducts",
+      "ssm:GetParameter",
     ]
-
-    resources = ["*"]
   }
 
   statement {
-    actions = [
-      "ec2:TerminateInstances",
-      "ec2:DeleteLaunchTemplate",
```

Comment:
we need to provide support for deleting launch templates that karpenter creates

Reply:
That's not required from my experience.